aws/request.go

```diff
@@ -24,6 +24,47 @@ const (
 	ErrCodeRead = "ReadError"
 )
 
+// HTTPSerializationError provides a generic request serialization error
+type HTTPSerializationError struct {
+	Reason string // provides more context of error caused
+	Err    error  // original error
+}
+
+// Error returns a formatted error for HTTPSerializationError
+func (e *HTTPSerializationError) Error() string {
+	return fmt.Sprintf("serialization failed: %v %v", e.Reason, e.Err)
+}
+
+// Unwrap returns the underlying Error in HTTPDeserializationError
+func (e *HTTPSerializationError) Unwrap() error { return e.Err }
+
+// HTTPDeserializationError provides a generic request deserialization error
+type HTTPDeserializationError struct {
+	Response  *http.Response
+	RequestID string // Aws specific info
+	Reason    string // provides more context of error caused
+	Err       error  //  original error
+}
+
+// Error returns a formatted error for HTTPDeserializationError
+func (e *HTTPDeserializationError) Error() string {
+	return fmt.Sprintf("deserialization failed, %v, %v", e.Reason, e.Err)
+}
+
+// ErrorStatusCode returns the Response status code
+func (e *HTTPDeserializationError) ErrorStatusCode() int {
+	if e.Response != nil {
+		return e.Response.StatusCode
+	}
+	return 0
+}
+
+// ErrorRequestID returns the Request ID for error response request
+func (e *HTTPDeserializationError) ErrorRequestID() string { return e.RequestID }
```

Comment:
These methods should just be `StatusCode`, and `RequestID` to conform to the current `awserr.RequestFailure` interface. Will have to address the member name collision though. Maybe use the v1 sdk's idea of a `protocol.ResponseMetadata` type as a member instead of the two individual members here. Poterntially adding `http.Response` to that type as well.